pkg/storage/storagewrappers/boundedconcurrency.go

```diff
@@ -0,0 +1,86 @@
+package storagewrappers
+
+import (
+	"context"
+	"time"
+
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const TimeWaitingSpanAttribute = "time_waiting"
+
+var _ storage.RelationshipTupleReader = (*boundedConcurrencyTupleReader)(nil)
+
+var (
+	timeWaitingCounter = promauto.NewHistogram(prometheus.HistogramOpts{
```

Comment:
This variable name should be a `___Histogram` not `___Counter`

Reply:
😅 oops haha